pkg/monitoring/metrics/virt-controller/vmstats_collector_test.go

```diff
@@ -346,7 +368,29 @@ var _ = Describe("VM Stats Collector", func() {
 			Expect(crs[1].Value).To(BeEquivalentTo(2097152))
 			Expect(crs[1].Labels).To(Equal([]string{"testvm", "test-ns", "memory", "bytes", "hugepages"}))
 
-			expectDefaultCPUResourceRequests(crs[2:])
+			// Find guest_effective memory requests in the results
+			var allocatedRequestsFound bool
+			for _, cr := range crs {
+				if len(cr.Labels) == 5 && cr.Labels[4] == "guest_effective" && cr.Labels[2] == "memory" {
+					if cr.Metric.GetOpts().Name == "kubevirt_vm_resource_requests" {
+						By("checking the guest_effective memory requests")
+						Expect(cr.Value).To(BeEquivalentTo(1024))
+						Expect(cr.Labels).To(Equal([]string{"testvm", "test-ns", "memory", "bytes", "guest_effective"}))
+						allocatedRequestsFound = true
+						break
+					}
+				}
+			}
+			Expect(allocatedRequestsFound).To(BeTrue(), "Should find guest_effective memory requests")
+
+			// Find and validate CPU default metrics
+			var cpuDefaultMetrics []operatormetrics.CollectorResult
+			for _, cr := range crs {
+				if len(cr.Labels) == 5 && cr.Labels[4] == "default" && cr.Labels[2] == "cpu" {
+					cpuDefaultMetrics = append(cpuDefaultMetrics, cr)
+				}
+			}
+			expectDefaultCPUResourceRequests(cpuDefaultMetrics)
```

Comment:
can we skip all this logic here, since we already have test below for the guest effective?
if not can we move it to a function

and the verification has a lot of identation and uses a lot of labels indexes, if you can hide them in functions like "has guest effective memory resource request"

Reply:
Updated